test/e2e/jobs/jobs.test.ts

```diff
@@ -11,18 +11,139 @@ See the License for the specific language governing permissions and
+            "--listen-address", "0.0.0.0",
+            "--port", "8083",
+            "--healthz-port", "8084",
+            "--enable-metrics", "false",
+            // note: This is here because `--enable-metrics=false` doesn't seem to be working.
+            "--metrics-port", "8085",
+          ])
+          // note: This is here because of daemon visibility issues.
+          .withNetworkMode("host")
+          .withTmpFs({
+            "/data": "rw",
+          })
+          // note: Because dapr containers don't have `sh` or `bash` inside, this is kind of the best health check.
+          .withWaitStrategy(Wait.forLogMessage("api is ready").withStartupTimeout(10000))
+          .start();
+
+        daprd = await new GenericContainer("ghcr.io/dapr/dapr")
+          .withExposedPorts(
+            {
+                host: 8081,
+                container: 8081
+            },
+            {
+                host: 8082,
+                container: 8082,
+            }
+          )
+          .withCommand([
+            "./daprd",
+            "--app-id", "dapr-js-sdk-testing",
+            "--app-port", "8070",
+            "--dapr-grpc-port", "8081",
+            "--dapr-http-port", "8082",
+            "--scheduler-host-address", "0.0.0.0:8083",
+            "--placement-host-address", "",
+            "--enable-metrics", "false",
+          ])
+          // note: This is here because well, the daemon likes it.
+          .withNetworkMode("host")
+          // note: Because dapr containers don't have `sh` or `bash` inside, this is kind of the best health check.
+          .withWaitStrategy(Wait.forLogMessage("HTTP server is running on port").withStartupTimeout(10000))
+          .start()
+        ;
+    });
 
     beforeEach(async () => {
 
+      server = new DaprServer({
+        serverHost: "127.0.0.1",
+        serverPort: "8070",
+        communicationProtocol: CommunicationProtocolEnum.HTTP,
+        clientOptions: {
+          daprHost: "127.0.0.1",
+          daprPort: "8082",
+        },
+      });
+
+      client = new DaprClient({
+        daprHost: "127.0.0.1",
+        daprPort: "8082",
+        communicationProtocol: CommunicationProtocolEnum.HTTP,
+      });
+
+      await server.start();
+    })
+
+    afterEach(async () => {
+
+      await client?.jobs.delete("test");
+      await server?.stop();
+      // await daprScheduler?.restart();
 
+      server = null;
+    })
+
+    afterAll(async () => {
+        await daprScheduler?.stop();
+        await daprd?.stop();
     });
 
+    it("Registers and receives a one second job five times.", async () => {
+
+        const callback = jest.fn(async () => { return null; });
+
+        server?.jobs.listen("test", callback);
+
+        await client?.jobs.schedule(
+          "test",
+          { value: "test" },
+          "* * * * * *"
+        );
+
+        const job = await client?.jobs.get("test");
+
+        await new Promise(resolve => setTimeout(resolve, 5000));
```

Comment:
The scheduler does not guarantee strict and immediate invocation, just that it'll be near the trigger point. Setting a timeout to 5000 ms suggests it's unlikely you'll see this invoked more than 4 times as there won't necessarily be strictly 1000 ms between each invocation.

I'd put the timeout at 6 seconds and measure for 5 invocations. Also, limit the repeats to 5 so the job cleans itself up.